components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java

```diff
@@ -0,0 +1,218 @@
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import static org.mockito.Matchers.anyString;
+import static org.powermock.api.mockito.PowerMockito.*;
+
+/**
+ * Unit test cases for the OpenID Connect Authenticator.
+ */
+@PrepareForTest({IdentityTenantUtil.class, BasicAuthenticatorServiceComponent.class, User
+        .class, MultitenantUtils.class, FrameworkUtils.class, FileBasedConfigurationBuilder.class,
+        IdentityUtil.class, UserCoreUtil.class})
+public class BasicAuthenticatorTestCase extends PowerMockTestCase {
+
+    private HttpServletRequest mockRequest;
+    private HttpServletResponse mockResponse;
+    private AuthenticationContext mockAuthnCtxt;
+    private RealmService mockRealmService;
+    private UserRealm mockRealm;
+    private UserStoreManager mockUserStoreManager;
+    private AbstractApplicationAuthenticator mockAbstractApplicationAuthenticator;
+    private FileBasedConfigurationBuilder mockFileBasedConfigurationBuilder;
+
+    @Mock
+    private User user;
+
+    private BasicAuthenticator basicAuthenticator;
+
+    @BeforeTest
+    public void setup() {
+
+        basicAuthenticator = new BasicAuthenticator();
+
+    }
+
+    @Test
+    public void canHandlePassTestCase() {
+
+        mockRequest = mock(HttpServletRequest.class);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.USER_NAME)).thenReturn("admin");
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.PASSWORD)).thenReturn("admin");
+        Assert.assertTrue(basicAuthenticator.canHandle(mockRequest), "Invalid can handle response for the request.");
+    }
+
+    @Test
+    public void canHandleFailUserNamePasswordTestCase() {
+
+        mockRequest = mock(HttpServletRequest.class);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.USER_NAME)).thenReturn(null);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.PASSWORD)).thenReturn(null);
+        Assert.assertFalse(basicAuthenticator.canHandle(mockRequest), "Invalid can handle response for the request.");
+    }
+
+    @Test
+    public void canHandleFaiUsernameTestCase() {
+
+        mockRequest = mock(HttpServletRequest.class);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.USER_NAME)).thenReturn(null);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.PASSWORD)).thenReturn("admin");
+        Assert.assertFalse(basicAuthenticator.canHandle(mockRequest), "Invalid can handle response for the request.");
+    }
+
+    @Test
+    public void canHandleFaiPasswordTestCase() {
+
+        mockRequest = mock(HttpServletRequest.class);
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.USER_NAME)).thenReturn("admin");
+        when(mockRequest.getParameter(BasicAuthenticatorConstants.PASSWORD)).thenReturn(null);
+        Assert.assertFalse(basicAuthenticator.canHandle(mockRequest), "Invalid can handle response for the request.");
```

Comment:
Use a data provider without repeating same logic with different params for all the above tests